### DISCUSSION TITLE
backreference through indirect multiple relation

### DISCUSSION THREAD

**Discussion by stuartin** [pocketbase/pocketbase#5123] [Q&A] [answered]
Hi,
I have the below relations setup between 4 collections:

![image](https://github.com/pocketbase/pocketbase/assets/13344713/2fe123fb-efa5-4fc2-a0bd-f10ddf29a728)
- Everthing has `single` relation to `toolbox`
- `tools` has `multiple` relation to `spanners`
- `sizes` has `single` relation to `spanners`

I'm trying to perform a backreference on the `tools` collection to expand the `sizes` collection.

The psuedo output I'm looking for is
```ts
tools = [
  {
    id: "xxx",
    ...,
    spanners: [
      {
        id: "xxx",
        ...,
        sizes: [
          {
            id: "xxx",
            ...
          },
        ],
      },
    ],
  },
];
```

I've tried setting `expand` to `sizes_via_spanners.id` and a few variations but cant seem to get anything returned.

Is this possible?

**Comment by ganigeorgiev** [comment, marked as answer]
> I'm trying to perform a backreference on the tools collection to expand the sizes collection.

I'm not sure I follow. `sizes_via_spanners.id` doesn't make sense as an `expand` value because `expand` expects only `relation` fields (or back-relation references). If there is nesting, the nested field must be also either a `relation` or back-relation reference.

Based on the pseudo-output I assume that the query you are looking for is probably:

```js
const tools = await pb.collection("tools").getFullList({
    expand: "spanners.sizes_via_spanner"
})
```

Also make sure that your client request satisfy the View API rule of the `spanners` and `sizes` collections.

_If the above doesn't help, please export your collections configuration as json (Admin UI > Settings > Export collections)._

**Reply by stuartin** [reply]
Nailed it as usual, thanks @ganigeorgiev, exactly what I was looking for. Much appreciated 🚀
